packages/gatsby/src/internal-plugins/dev-404-page/raw_dev-404-page.js

```diff
@@ -243,17 +243,20 @@ export default function API (req, res) {
 
 export default Dev404Page
 
+// ESLint is complaining about the backslash in regex
+/* eslint-disable */
 export const pagesQuery = graphql`
   query PagesQuery {
     allSiteFunction {
       nodes {
         functionRoute
       }
     }
-    allSitePage(filter: { path: { ne: "/dev-404-page/" } }) {
+    allSitePage(filter: { path: { regex: "/^(?!\/dev-404-page).+$/" } }) {
```

Comment:
Do we also use this in sitemap plugin or other plugins where we'd need to adjust this?